src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -305,16 +356,90 @@ bool IsCreateHelper(IMethodSymbol method)
                     method.Parameters[0].Type.SpecialType == SpecialType.System_String;
         }
 
-        private static void AnalyzeTitle(IMethodSymbol descriptorCreationMethod, IFieldInitializerOperation creation, Action<Diagnostic> reportDiagnostic)
+        private static void AnalyzeTitle(OperationAnalysisContext operationAnalysisContext, ImmutableArray<IArgumentOperation> creationArguments, IFieldInitializerOperation creation)
+        {
+            IArgumentOperation titleArgument = creationArguments.FirstOrDefault(a => a.Parameter.Name.Equals("title", StringComparison.OrdinalIgnoreCase));
+            if (titleArgument?.Parameter.Type.SpecialType != SpecialType.System_String)
+            {
+                return;
+            }
+
+            operationAnalysisContext.ReportDiagnostic(creation.Value.CreateDiagnostic(UseLocalizableStringsInDescriptorRule, WellKnownTypeNames.MicrosoftCodeAnalysisLocalizableString));
+
+            if (!TryGetNonEmptyConstantStringValue(titleArgument, out var title))
+            {
+                return;
+            }
+
+            if (IsMultiSentences(title) || EndsWithPeriod(title) || ContainsLineReturn(title))
+            {
+                operationAnalysisContext.ReportDiagnostic(titleArgument.CreateDiagnostic(DefineDiagnosticTitleCorrectlyRule));
+            }
+        }
+
+        private static void AnalyzeMessage(OperationAnalysisContext operationAnalysisContext, ImmutableArray<IArgumentOperation> creationArguments)
         {
-            IParameterSymbol title = descriptorCreationMethod.Parameters.FirstOrDefault(p => p.Name == "title");
-            if (title != null &&
-                title.Type != null &&
-                title.Type.SpecialType == SpecialType.System_String)
+            IArgumentOperation messageArgument = creationArguments.FirstOrDefault(a => a.Parameter.Name.Equals("messageFormat", StringComparison.OrdinalIgnoreCase));
+            if (!TryGetNonEmptyConstantStringValue(messageArgument, out var message))
             {
-                Diagnostic diagnostic = creation.Value.CreateDiagnostic(UseLocalizableStringsInDescriptorRule, WellKnownTypeNames.MicrosoftCodeAnalysisLocalizableString);
-                reportDiagnostic(diagnostic);
+                return;
             }
+
+            var isMultiSentenceMessage = message.Contains(". ");
```

Comment:
```suggestion
```

Reply:
I can handle this as part of the follow-up code fix PR.

Reply:
Sorry I missed it!